src/text.rs

```diff
@@ -0,0 +1,137 @@
+use std::cmp::PartialEq;
+use std::convert::{AsRef, From};
+use std::io::{BufRead, Write};
+use std::ops::Deref;
+
+use quick_xml::events::attributes::Attributes;
+use quick_xml::events::{BytesEnd, BytesStart, BytesText, Event};
+use quick_xml::Error as XmlError;
+use quick_xml::Reader;
+use quick_xml::Writer;
+
+use crate::error::Error;
+use crate::fromxml::FromXml;
+use crate::toxml::ToXmlNamed;
+use crate::util::{atom_text, atom_xhtml};
+
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[derive(Debug, Clone, PartialEq, Default)]
+/// Represents a [text construct](https://tools.ietf.org/html/rfc4287#section-3.1) in an Atom feed.
+pub struct Text {
+    /// Content of the text construct
+    pub value: String,
+    /// Base URL for resolving any relative references found in the element.
+    pub base: Option<String>,
+    /// Indicates the natural language for the element.
+    pub lang: Option<String>,
+    /// Type of content stored in the element.
+    /// When present, the value MUST be one of "text", "html", or "xhtml".
```

Comment:
This is not completely correct, tho, alternatively, it could also be a MIME type.

Reply:
I'm not sure how it is handled in other parts of the library, but it might be a good idea to replace the value type of `content_type` with an enum:
```rust
enum ContentType {
    Text,
    Html,
    Xhtml,
    Mime(String),
}
```

Reply:
And only output it if the `ContentType` is not the default value.

Reply:
[spec](https://tools.ietf.org/html/rfc4287#section-3.1) says:
> When present, the value MUST be one of "text", "html", or "xhtml".

Reply:
oh ok, I might have confused something...